examples/threathunter/threat_intelligence/stix_taxii.py

```diff
@@ -0,0 +1,270 @@
+            logging.warning(f"Problem fetching collections from taxii server: {e}")
+        return collections
+
+    def poll_server(self, collection, feed_helper):
+        content_blocks = []
+        uri = self.create_uri(self.config.poll_path)
+        try:
+            logging.info(f"Polling Collection: {collection.name}")
+            content_blocks = self.client.poll(
+                uri=uri,
+                collection_name=collection.name,
+                begin_date=feed_helper.start_date,
+                end_date=feed_helper.end_date,
+                content_bindings=BINDING_CHOICES)
+        except handled_exceptions as e:
+            logging.warning(f"problem polling taxii server: {e}")
+        return content_blocks
+
+    def parse_collection_content(self, content_blocks):
+        for block in content_blocks:
+            yield from parse_stix(block.content, self.config.default_score)
+
+    def import_collection(self, collection):
+        num_times_empty_content_blocks = 0
+        advance = True
+        reports_limit = self.config.reports_limit
+        logging.info(f"reports limit: {reports_limit}")
+        feed_helper = FeedHelper(self.config.start_date,
+                                 self.config.minutes_to_advance)
+
+        # config parameters `start_date` and `minutes_to_advance` tell this Feed Helper
+        # where to start polling in the collection, and then will advance polling in chunks of
+        # `minutes_to_advance` minutes until we hit the most current `content_block`.
+        while feed_helper.advance():
+            num_reports = 0
+            content_blocks = self.poll_server(collection, feed_helper)
+            reports = self.parse_collection_content(content_blocks)
+            for report in reports:
+                yield report
+                num_reports += 1
+                if reports_limit is not None and num_reports > reports_limit:
+                    logging.info(f"Reports limit of {reports_limit} reached")
+                    advance = False
+                    break
+
+            if not advance:
+                break
+            if collection.type == 'DATA_SET':  # data is unordered, not a feed
+                logging.info(f"collection:{collection}; type data_set; breaking")
+                break
+            if num_reports == 0:
+                num_times_empty_content_blocks += 1
+            if num_times_empty_content_blocks > self.config.fail_limit:
+                logging.error('Max fail limit reached; Exiting.')
+                break
+            if reports_limit is not None:
+                reports_limit -= num_reports
+
+    def import_collections(self, available_collections):
+        desired_collections = self.config.collections
+        desired_collections = [x.strip()
+                               for x in desired_collections.lower().split(',')]
+
+        want_all = False
+        if '*' in desired_collections:
+            want_all = True
```

Comment:
this would be better as a ternary statement (reduce mutability as much as possible).

Reply:
changed to `want_all = True if '*' in desired_collections else False`